ignite/contrib/handlers/tensorboard_logger.py

```diff
@@ -522,39 +538,78 @@ class GradsScalarHandler(BaseWeightsScalarHandler):
             # Create a logger
             tb_logger = TensorboardLogger(log_dir="experiments/tb_logs")
 
-            # Attach the logger to the trainer to log model's weights norm after each iteration
+            # Attach the logger to the trainer to log model's gradients norm after each iteration
             tb_logger.attach(
                 trainer,
                 event_name=Events.ITERATION_COMPLETED,
                 log_handler=GradsScalarHandler(model, reduction=torch.norm)
             )
-    """
 
-    def __init__(self, model: nn.Module, reduction: Callable = torch.norm, tag: Optional[str] = None):
-        super(GradsScalarHandler, self).__init__(model, reduction, tag=tag)
+        .. code-block:: python
+
+            from ignite.contrib.handlers.tensorboard_logger import *
+
+            tb_logger = TensorboardLogger(log_dir="experiments/tb_logs")
+
+            # Log gradient of `base`
+            tb_logger.attach(
+                trainer,
+                event_name=Events.ITERATION_COMPLETED,
+                log_handler=GradsScalarHandler(
+                    model,
+                    reduction=torch.norm,
+                    whitelist=['base']
+                )
+            )
+
+        .. code-block:: python
+
+            from ignite.contrib.handlers.tensorboard_logger import *
+
+            tb_logger = TensorboardLogger(log_dir="experiments/tb_logs")
+
+            # Log gradient of weights which belong to a `fc` layer
+            def is_in_fc_layer(n, p):
+                return 'fc' in n
+
+            tb_logger.attach(
+                trainer,
+                event_name=Events.ITERATION_COMPLETED,
+                log_handler=GradsScalarHandler(model, whitelist=is_in_fc_layer)
+            )
+
+    ..  versionchanged:: 0.5.0
+        optional argument `whitelist` added.
+    """
 
     def __call__(self, engine: Engine, logger: TensorboardLogger, event_name: Union[str, Events]) -> None:
         if not isinstance(logger, TensorboardLogger):
             raise RuntimeError("Handler 'GradsScalarHandler' works only with TensorboardLogger")
 
         global_step = engine.state.get_event_attrib_value(event_name)
         tag_prefix = f"{self.tag}/" if self.tag else ""
-        for name, p in self.model.named_parameters():
+        for name, p in self.weights:
             if p.grad is None:
                 continue
 
             name = name.replace(".", "/")
             logger.writer.add_scalar(
-                f"{tag_prefix}grads_{self.reduction.__name__}/{name}", self.reduction(p.grad), global_step
+                f"{tag_prefix}grads_{self.reduction.__name__}/{name}", self.reduction(p.grad).item(), global_step
```

Comment:
`self.reduction(p.grad)` is not always Tensor and can be also float. Thus calling `.item()` will fail. I think it is safe to remove `.item` here.